sdk/iot/Azure.IoT.Hub.Service/tests/QueryTests.cs

```diff
@@ -127,8 +131,11 @@ await foreach (TwinData item in queryResponse)
 
                     tryCount++;
 
-                    // Adding a delay to account for query cache sync.
-                    await Task.Delay(TimeSpan.FromSeconds(5)).ConfigureAwait(false);
+                    if (Mode != RecordedTestMode.Playback)
+                    {
+                        // Adding a delay to account for query cache sync.
+                        await Task.Delay(TimeSpan.FromSeconds(5)).ConfigureAwait(false);
```

Comment:
Do we need a helper for this in the TestFrameowrk? RecordedTestBase.Delay?

Reply:
@heaths Suggested the same thing. I think it's a good idea, but seemed rare enough in this PR not to be a big deal. Perhaps I'll do it.

Reply:
It seems to be the most common pattern to slow tests down.

Reply:
I've implemented it, just forgot to use it here.